### PR TITLE
Update Anonymous telemetry metrics according to metrics naming guidelines

### PR DESCRIPTION
**Pull request by sgrebnov**
## 🗣 Description

Update Anonymous telemetry metrics according to [Metrics naming principles and guidelines](https://github.com/spiceai/spiceai/pull/3516)

Dashboards will be updated/tested separately

**This is a Breaking Change**, changes are tracked as part of https://github.com/spiceai/spiceai/issues/3450

## 🔨 Related Issues

Part of https://github.com/spiceai/spiceai/issues/3450

### CODE REVIEW

File: crates/runtime/src/metrics.rs [hunk 1]
```diff
@@ -269,7 +269,7 @@ pub(crate) mod telemetry {
         TELEMETRY_METER
             .u64_counter("query_processed_bytes")
             .with_description("Number of bytes processed by the runtime.")
-            .with_unit("bytes")
+            .with_unit("By")
```
Comment (sgrebnov) [inline crates/runtime/src/metrics.rs:272]: This is unrelated change, added for consistency and to match proposal to follow UCUM units

**Review by phillipleblanc — APPROVED** [review]
